go/libkb/skb.go

```diff
@@ -306,6 +303,12 @@ func (s *SKB) UnlockSecretKey(lctx LoginContext, passphrase string, tsec Triples
 	default:
 		err = BadKeyError{fmt.Sprintf("Can't unlock secret with protection type %d", int(s.Priv.Encryption))}
 	}
+
+	if key = s.decryptedSecret; key != nil {
+		// Key is already unlocked and unpacked, do not parse again.
+		return
+	}
+
```

Comment:
I can't say I was able to check everything and make sure... So I'll be fine with working around this in pgp engine.